grid-client/deployer/network_deployer.go

```diff
@@ -318,36 +432,29 @@ func (d *NetworkDeployer) Deploy(ctx context.Context, znet *workloads.ZNet) erro
 
 // BatchDeploy deploys multiple network deployments using the deployer
 func (d *NetworkDeployer) BatchDeploy(ctx context.Context, znets []*workloads.ZNet, updateMetadata ...bool) error {
-	newDeployments := make(map[uint32][]gridtypes.Deployment)
-	newDeploymentsSolutionProvider := make(map[uint32][]*uint64)
-	nodePorts := make(map[uint32][]uint16)
+	filteredZNets := make([]*workloads.ZNet, 0)
 	var multiErr error
+	if err := validateAccountBalanceForExtrinsics(d.tfPluginClient.SubstrateConn, d.tfPluginClient.Identity); err != nil {
+		return err
+	}
 	for _, znet := range znets {
-		err := d.Validate(ctx, znet)
-		if err != nil {
-			multiErr = multierror.Append(multiErr, fmt.Errorf("invalid network %s: %w", znet.Name, err))
+		if err := znet.Validate(); err != nil {
```

Comment:
we can refactor the validate function of the network deployer for multiple znets